### DISCUSSION TITLE
Surface normal orientation

### DISCUSSION THREAD

**Discussion by csparker247** [educelab/volume-cartographer#50] [Ideas]
The surface normals for segmented surfaces are determined primarily by the connectedness of the original segmentation. That is, if you draw an initial segmentation from left-to-right, it will have inverse normals of the same segmentation drawn from right-to-left. In essence, the sign of the surface normal is arbitrary geometrically.

It would be good to auto-orient the normals in a consistent manner within the same volume so that things like `vc_layers` operate in consistent orientations with no manual intervention. The proposal in #48 uses the mean of the segmentation point set to determine the sign of the segmentation normals, which is probably good for convex surfaces (i.e. scrolls) but maybe not so good for flat ones (books, codices, etc.) Still, I think we can generalize the concept by allowing similar auto-orientation w.r.t. volume landmarks, like the axes of the bounding box.

**Comment by spelufo** [comment]
I'd say forget about the original axis. They have no connection with the desired target coordinate system. If you scan a scroll or other object in another orientation the relationship of the object to the scan's axis is arbitrary. Some user defined landmarks like the scroll core are needed. Until that is thought out and implemented I think the convex proposal would be good to have for the segments that are being produced and used in the vesuvius challenge. If the bounding box is available, the center point of that would also serve as a reference point for the normal (probably worse results, but cheaper to compute). Do the simple thing first, it might turn out to be the right thing after all :).

**Reply by csparker247** [reply]
I think the significance of the axes of the original scan are unknown at first, but that doesn't mean they're not meaningful with respect to the desired coordinate system. For example, the XZ planes (top and bottom edges of the slices) of the volume bounding box are very good orientation markers for the fragment scans, which have a clear front and back. The same is true for books we have scanned.

I'm not saying I'm not going to merge the convex idea. I just think it's geared towards scrolls, and that there are very simple extensions that will meet other use cases.

**Comment by csparker247** [comment]
Fixed in #48.
